ckanext/validation/logic.py

```diff
@@ -672,3 +687,13 @@ def _run_sync_validation(resource_id, local_upload=False, new_resource=True):
 
         raise t.ValidationError({
             u'validation': [report]})
+
+
+@t.chained_action
+def package_patch(original_action, context, data_dict):
+    ''' Detect whether resources have been replaced, and if not,
+    place a flag in the context accordingly.
+    '''
+    if 'resources' not in data_dict:
+        context['save'] = True
```

Comment:
nice find, context for update/new from user would have this. i.e.
``` context['save'] = 'save' in request.params```

```
          context = {'model': model, 'session': model.Session,
                   'user': c.user,
                   'save': 'save' in request.params,
                   'parent': request.params.get('parent', None)}
```
this behaviour is in:
* ckan.controllers.package.new 
* ckan.controllers.package.edit
* ckan.controllers.user.new
* ckan.controllers.user.edit
* ckan.controllers.group.new
* ckan.controllers.group.edit


there is also flag
```context['for_edit'] = True```

Reply:
Yes, the fact that it's an existing flag is why I used it. It seems to be perfect for distinguishing between updates via the web form (which by their nature don't affect resources) vs API calls (which might).